asserts/repair.go

```diff
@@ -0,0 +1,81 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package asserts
+
+import (
+	"fmt"
+)
+
+// Repair holds an repair assertion which allows running repair
+// code to fixup broken systems. It can be limited by series and models.
+type Repair struct {
+	assertionBase
+
+	series []string
+	models []string
+}
+
+// RepairID returns the "id" of the repair. It should be a short string
+// that follows a convention like "REPAIR-123". Similar to a CVE there
+// should be a public place to look up details about the repair-id
+// (e.g. the snapcraft forum).
+func (em *Repair) RepairID() string {
```

Comment:
We can either have a single ID, or a split one ("REPAIR", 123).  Having a single one is probably a way to reinforce the idea of such IDs, so seems nice, and either way we'll end up with that pair as the primary key, so not much different.

If keeping it as a single string, we should probably make sure it conforms to the format `<key>-<num>` though, so that we can take some further decisions based on that. One important behavior to establish is ordering. We probably don't want arbitrary ordering, but at the same time we probably don't want unrelated patch sets to depend on each other. Instead, we should probably sequence entries with the same `<key>` to ensure the order of `<num>` is respected, applying one after another, while accepting independent keys to be applied out of order.

Comments? Ideas?